tests/scripts/run-xunit-slice.py

```diff
@@ -0,0 +1,82 @@
+import json
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+
+##########################################################################
+# Argument Parser
+##########################################################################
+
+description = 'Tool to generate slice<number>.txt files based off of a slice json file (see perf-slices.json for format)'
+
+parser = argparse.ArgumentParser(description=description)
+
+parser.add_argument('-slice', dest='slice_number', default=-1, type=int)
+parser.add_argument('-slice-file', dest='slice_file', default=None)
+parser.add_argument('-slice-dir', dest='slice_dir', default=None)
+
+##########################################################################
+# Helper Functions
+##########################################################################
+
+def validate_args(args):
+    """ Validate all of the arguments parsed.
+    Args:
+        args (argparser.ArgumentParser): Args parsed by the argument parser.
+    Returns:
+        (arch, build_type, clr_root, fx_root, fx_branch, fx_commit, env_script)
+            (str, str, str, str, str, str, str)
+    Notes:
+    If the arguments are valid then return them all in a tuple. If not, raise
+    an exception stating x argument is incorrect.
+    """
+
+    slice_number = args.slice_number
+    slice_file = args.slice_file
+    slice_dir = args.slice_dir
+
+    def validate_arg(arg, check):
+        """ Validate an individual arg
+        Args:
+           arg (str|bool): argument to be validated
+           check (lambda: x-> bool): test that returns either True or False
+                                   : based on whether the check passes.
+
+        Returns:
+           is_valid (bool): Is the argument valid?
+        """
+
+        helper = lambda item: item is not None and check(item)
+
+        if not helper(arg):
```

Comment:
This could be more clear if you take the logic out of the lambda and place it in the if statement instead.  Because you aren't passing it out of this context, I don't see why the lambda is needed

Reply:
Again, this was copied from other python scripts. I'd just rather not have to rewrite our arg validation code, and would prefer it match the others.